### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by marcocalignano**
- Multigame mode update the new downloaded network on each thread. 
- Clean exit after validation mode is terminated

### CODE REVIEW

**Comment by gcp** [comment]
Awesome! With these fixes and a the resigning stuff I think we can consider a new release.

**Comment by marcocalignano** [comment]
If you do a new release the next 'next' branch I can give all the work on the distributed verification that I am working on with @roy7 so people can start testing it.

File: autogtp/Production.h [hunk 1]
```diff
@@ -30,19 +30,25 @@
...
+        NET_CHANGE,
+        FINISHING        
+    };
     ProductionWorker() = default;
     ProductionWorker(const ProductionWorker& w) : QThread(w.parent()) {}
     ~ProductionWorker() = default;
-    void init(const QString& gpuIndex, const QString& net, const int index);
-    void newNetwork(const QString& net) { m_network = net; }
+    void init(const QString& gpuIndex, const QString& net);
+    void newNetwork(const QString& net) { QMutexLocker locker(&m_mutex); m_state = NET_CHANGE; m_network = net; }
```
Comment (sethtroisi) [inline autogtp/Production.h]: nit: on newlines

File: autogtp/Production.h [hunk 2]
```diff
@@ -30,19 +30,25 @@
...
     void run() override;
-
+
```
Comment (sethtroisi) [inline autogtp/Production.h]: nit: remove trailing whitespace

File: autogtp/Production.cpp [hunk 3]
```diff
@@ -41,29 +43,49 @@ void ProductionWorker::run() {
...
-         QTextStream(stdout) << "Game has ended." << endl;
-         if (game.getScore()) {
-             game.writeSgf();
-             game.dumpTraining();
+         } while (game.nextMove() && m_state == RUNNING);
+         switch(m_state) {
+         case RUNNING:
+         {
+             QTextStream(stdout) << "Game has ended." << endl;
+             if (game.getScore()) {
```
Comment (sethtroisi) [inline autogtp/Production.cpp]: Here you only writeSgf if (getScore)
on line 64 you always write the file,

should they use the same logic?

File: autogtp/Validation.cpp [hunk 4]
```diff
@@ -50,44 +50,49 @@ void ValidationWorker::run() {
...
-        net = m_secondNet;
-        m_secondNet = m_firstNet;
-        m_firstNet = net;
-        if(m_expected == Game::BLACK) {
-            m_expected = Game::WHITE;
-        } else {
-            m_expected = Game::BLACK;
         }
-    } while (1);
+
```
Comment (sethtroisi) [inline autogtp/Validation.cpp]: nit: rm newlines

**Comment by gcp** [comment]
I fixed the style nits in d8eef505d46e5de58120662163bd124b7fc88f9a, remains the question about the if (getScore) logic. Note that  #113 got blocked on a similar issue.

**Comment by marcocalignano** [comment]
@sethtroisi you are right in Valdation.cpp the all  `if(!m_keepPath.isEmpty())` 
block has to go inside the `if (game.getScore())` block

**Pull request merged**